src/regtech_data_validator/check_functions.py

```diff
@@ -318,7 +318,18 @@ def is_number(ct_value: str, accept_blank: bool = False) -> bool:
     Returns:
         bool: True if value is number , False if value is not number
     """
-    value_check = ct_value.isdigit() or bool(re.match(r"^[-+]?[0-9]*\.?[0-9]+$", ct_value))
+
+    def num_check():
```

Comment:
if we want to abstract the logic into its own function, I'd rather see it outside of the `is_number` function